aiida_pseudo/cli/install.py

```diff
@@ -23,43 +24,56 @@ def cmd_install():
+    `archive_or_folder` is assumed to be an archive and the command will attempt to infer the archive format from the
+    filename extension of the ARCHIVE. If this fails, the archive format can be specified explicitly with the archive
+    format option, which will also display which formats are supported.
 
-    By default, the command will create a base `PseudoPotentialFamily`, but the type can be changed with the family
+    By default, the command will create a base `PseudoPotentialFamily`, but the type can be changed with the pseudos
     type option. If the base type is used, the pseudo potential files in the archive *have* to have filenames that
     strictly follow the format `ELEMENT.EXTENSION`, because otherwise the element cannot be determined automatically.
     """
     from .utils import attempt, create_family_from_archive
+    from aiida_pseudo.groups.family.pseudo import PseudoPotentialFamily
 
-    # The `archive` is now either a `http.client.HTTPResponse` or a normal filelike object, so we get the original file
-    # name in a different way.
-    try:
-        suffix = os.path.basename(archive.url)
-    except AttributeError:
-        suffix = os.path.basename(archive.name)
-
-    # Write the content of the archive to a temporary file, because `create_family_from_archive` does currently not
-    # accept filelike objects because the underlying `shutil.unpack_archive` does not. Likewise, `unpack_archive` will
-    # attempt to deduce the archive format from the filename extension, so it is important we maintain the original
-    # filename. Of course if this fails, users can specify the archive format explicitly wiht the corresponding option.
-    with tempfile.NamedTemporaryFile(mode='w+b', suffix=suffix) as handle:
-        shutil.copyfileobj(archive, handle)
-        handle.flush()
+    # `archive_or_folder` is a simple string, containing the name of the folder / file / url location.
 
+    if os.path.isdir(archive_or_folder):
+        try:
+            family = PseudoPotentialFamily.create_from_folder(archive_or_folder, label, pseudo_type=pseudos_type)
+        except ValueError as exception:
+            raise OSError(f'failed to parse pseudos from `{archive_or_folder}`: {exception}') from exception
+    elif os.path.isfile(archive_or_folder):
+        archive = archive_or_folder
         with attempt('unpacking archive and parsing pseudos... ', include_traceback=traceback):
-            family = create_family_from_archive(family_type, label, pathlib.Path(handle.name), fmt=archive_format)
+            family = create_family_from_archive(
+                PseudoPotentialFamily, label, pathlib.Path(archive), fmt=archive_format, pseudo_type=pseudos_type
+            )
+    else:
+        # The file of the url must be copied to a local temporary file. Maybe better ways to do it?
+        with urllib.request.urlopen(archive_or_folder) as archive:
```

Comment:
You are not checking if the opening of the URL actually works and so this will just except if there is anything wrong (I realize the original code also didn't check but it would be good to add now). Should probably use the `requests` library instead which makes it easy to check for problems. See the `download_sssp` for an example.

Reply:
This is already checked by `PathOrUrl` I believe.